server/activate.go

```diff
@@ -24,89 +23,105 @@ func (p *Plugin) OnActivate() error {
 		p.serverVersion = serverVersion
 	}
 
-	if err := p.ensureBotExists(); err != nil {
+	if botUserId, err := p.ensureBotExists(); err != nil {
 		return errors.Wrap(err, "failed to ensure bot user exists")
+	} else {
+		p.botUserId = botUserId
 	}
 
 	p.initializeClient()
 
 	p.registerCommands()
 
+	p.setActivated(true)
```

Comment:
Is this the only call to `setActivated`? I /think/ `OnActivate` is only called once, and on startup: do we need the locking around the corresponding variable at all?

Reply:
That's true. I was probably overthinking the locking on that one since I was really concerned about races earlier.